### ISSUE TITLE
Roslyn doesn't warn about `default` default parameter value for generics that might be not nullable reference types

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#44341]
**Version Used**: 
```
Branch master (17 May 2020)
Latest commit f03f87f by Tomáš Matoušek:
Delete RemoteHostClientMock (#44318)
```

**Steps to Reproduce**:

Compile and run the following code:
```cs
#nullable enable
public class C {
    public static void Main()
    {
      var s = M<string>();
      s.ToString();
    }
    
    static T M<T>(T t = default) => t;
}
```

**Expected Behavior**:
Either warning for `T t = default` default paramete value or warning for `=> t` returning possible null value

**Actual Behavior**:
No warnings. The program crashes at runtime with a null reference exception

**Comment by RikkiGibson** [comment]
this seems specific to default parameter values. I wonder if there is some consolidation we could do to try and make default parameter value bugs less common.

```cs
#nullable enable
public class C {
    public static void Main()
    {
      var s = M<string>();
      s.ToString();
    }
    
    static T M<T>(T t = default) => t;  // no warning on 'default'
    
    static T M1<T>() => M<T>(default);  // warning on 'default'
}
```

**Comment by RikkiGibson** [comment]
Seems related to #43398

**Comment by RikkiGibson** [comment]
@jaredpar I swear you just tagged me on a bug very similar to this, but I can't find it now.

**Comment by jaredpar** [comment]
@RikkiGibson believe you are thinking about #40818

**Comment by RikkiGibson** [comment]
Closing as a duplicate of #40818
